_episodes/62-cd.md

```diff
@@ -0,0 +1,125 @@
+> 
+> image: node:16
+> 
+> # Workflow Configuration
+> 
+> pipelines:
+>   default:
+>     - parallel:
+>       - step:
+>           name: Build and Test
+>           caches:
+>             - node
+>           script:
+>             - npm install
+>             # CI=true in default variables for Bitbucket Pipelines https://support.atlassian.com/bitbucket-cloud/docs/variables-in-pipelines/
+>             - npm test
+>       - step:
+>           name: Lint the node package
+>           script:
+>             # Run your linter of choice here
+>             - npm install eslint
+>             - npx eslint src
+>           caches:
+>             - node
+>   branches:
+>     master:
+>       - parallel:
+>         - step:
+>             name: Build and Test
+>             caches:
+>               - node
+>             script:
+>               - npm install
+>               # CI=true in default variables for Bitbucket Pipelines https://support.atlassian.com/bitbucket-cloud/docs/variables-in-pipelines/
+>               - npm test
+>               - npm run build
+>             artifacts:
+>               - build/**
+>         - step:
+>             name: Security Scan
+>             script:
+>               # Run a security scan for sensitive data.
+>               # See more security tools at https://bitbucket.org/product/features/pipelines/integrations?&category=security
+>               - pipe: atlassian/git-secrets-scan:0.5.1
+>       - step:
+>           name: Deploy to Production
+>           deployment: Production
+>           trigger: manual
+>           clone:
+>             enabled: false
+>           script:
+>             # sync your files to S3
+>             - pipe: atlassian/aws-s3-deploy:1.1.0
+>               variables:
+>                 AWS_ACCESS_KEY_ID: $AWS_ACCESS_KEY_ID
+>                 AWS_SECRET_ACCESS_KEY: $AWS_SECRET_ACCESS_KEY
+>                 AWS_DEFAULT_REGION: $AWS_DEFAULT_REGION
+>                 S3_BUCKET: 'my-bucket-name'
+>                 LOCAL_PATH: 'build'
+>             # triggering a distribution invalidation to refresh the CDN caches
+>             - pipe: atlassian/aws-cloudfront-invalidate:0.6.0
+>               variables:
+>                 AWS_ACCESS_KEY_ID: $AWS_ACCESS_KEY_ID
+>                 AWS_SECRET_ACCESS_KEY: $AWS_SECRET_ACCESS_KEY
+>                 AWS_DEFAULT_REGION: $AWS_DEFAULT_REGION
+>                 DISTRIBUTION_ID: '123xyz'
+> ~~~
+> {: .language-yml}
+> 
+> > ### Solution
+> > 1. The 4 steps are: Build and Test, Lint the node package, Security Scan, and Deploy to Production.
+> > 2. The top 'Build and Test' step first installs the application, then tests it.
+> >    The bottom 'Build and Test' step also builds the application and stores the build in an artifact.
+> >    The top 'Build and Test' step is run in all branches apart from master. 
+> >    The bottom 'Build and Test' step is only run on the master branch.
```

Comment:
Why is there no linting on `main`? Because code comes in via merge requests from other branches where the linting already happened?

Reply:
Yes, that's the idea.